azerothcore/azerothcore-wotlk
# feat(APPS/installer): configure remote and pull branch in conf/config.sh

## Changes Proposed:
* Added `INSTALLER_PULL_FROM` in conf/config.sh.dist as PRed in https://github.com/azerothcore/azerothcore-wotlk/pull/3196.
  - The default value of `INSTALLER_PULL_FROM` is the currently checked out branch as [discussed](https://github.com/azerothcore/azerothcore-wotlk/pull/3196#discussion_r458575526) in 3196.

* Added `ORIGIN_REMOTE` in conf/config.sh.dist as requested in https://github.com/azerothcore/azerothcore-wotlk/issues/1064 to pull from azerothcore's official repo.
  - The default value of `ORIGIN_REMOTE` is https://github.com/azerothcore/azerothcore-wotlk.git

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1064

## Tests Performed:

OS: Linux Mint 20.1
AC commit/hash: https://github.com/azerothcore/azerothcore-wotlk/commit/d0ff62a1ab448d55509f7a48b864490c2ae9c137


## How to Test the Changes:


1. Change remote `origin` to something else, e.g. the address of your fork
   - check that `ORIGIN_REMOTE` in conf/config.sh is the default https://github.com/azerothcore/azerothcore-wotlk.git
2. Choose option `3)` of installer.sh
3. It will pull from azerothcore's official repo using the currently checked out branch

## Known Limitation:
If you are on a branch not present in the official repo, option `3)` of installer.sh will (obviously) fail to pull from the default `ORIGIN_REMOTE`


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
